data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/LocalModePeerForwarderProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.peerforwarder.RequiresPeerForwarding;
+
+import java.util.Map;
+import java.util.Set;
+
+public class LocalModePeerForwarderProvider implements PeerForwarderProvider {
+
+    private final PeerForwarderProvider peerForwarderProvider;
+    private boolean isRemotePeerForwarderRegistered;
+
+    public LocalModePeerForwarderProvider(final PeerForwarderProvider peerForwarderProvider) {
+        this.peerForwarderProvider = peerForwarderProvider;
+        this.isRemotePeerForwarderRegistered = false;
+    }
+
+    @Override
+    public PeerForwarder register(final String pipelineName, final Processor processor, final String pluginId, final Set<String> identificationKeys, final Integer pipelineWorkerThreads) {
+        if (((RequiresPeerForwarding)processor).isForLocalProcessingOnly(null)) {
+            return new LocalPeerForwarder();
+        }
+        isRemotePeerForwarderRegistered = true;
+        return peerForwarderProvider.register(pipelineName, processor, pluginId, identificationKeys, pipelineWorkerThreads);
+    }
+
+    @Override
+    public boolean isPeerForwardingRequired() {
+        return isRemotePeerForwarderRegistered;
```

Comment:
It is a little strange that there is no way to verify that `register` is called before this method, meaning that calling this method at first would return false, then after calling `register` it may return true. Not sure if it's worth finding a way to enforce this